src/elf.rs

```diff
@@ -727,83 +725,31 @@ impl<'a> Elf<'a> {
-            .section_index(shndx)
-            .create();
-
-        let final_symbol_count =
-            self.special_symbols.len() + self.sections.len() + self.symbols.len() + 1;
-        let section = SectionBuilder::new((final_symbol_count * 4) as u64)
-            .section_type(SectionType::SymTabShndx);
-
-        let mut section = section.name_offset(offset).create(&self.ctx);
-        section.sh_link = 2;
-        // the offset is the head of how many program bits we've added
-        section.sh_offset = self.sizeof_bits as u64;
-        self.sections.insert(
-            idx,
-            SectionInfo {
-                header: section,
-                symbol: section_symbol,
-                name: idx,
-            },
-        );
-
-        // now that we inserted the final symbol, build the actual table
-        let mut data =
-            vec![0u8; (self.special_symbols.len() + self.sections.len() + self.symbols.len()) * 4];
-        let mut offset = 0;
-
-        for symbol in &self.special_symbols {
-            data.gwrite_with(symbol.st_shndx as u32, &mut offset, self.ctx.le)
-                .expect("preallocated shndx vector has enough space for special symbols");
-        }
-        for (_id, section) in &self.sections {
-            data.gwrite_with(section.symbol.st_shndx as u32, &mut offset, self.ctx.le)
-                .expect("preallocated shndx vector has enough space for sections");
-        }
-        for (_id, symbol) in &self.symbols {
-            data.gwrite_with(symbol.st_shndx as u32, &mut offset, self.ctx.le)
-                .expect("preallocated shndx vector has enough space for symbols");
-        }
-
-        // store the size of this code
-        let size = data.len();
-        self.extended_symtab = Some(data);
-
-        self.nsections += 1;
-        // increment the size
-        self.sizeof_bits += size;
-        shndx
-    }
     fn align(offset: &mut u64, sizeof_t: u64) {
         let alignment = *offset % sizeof_t;
-        *offset += sizeof_t - alignment;
+        if alignment != 0 {
+            *offset += sizeof_t - alignment;
+        }
     }
     pub fn write<T: Write + Seek>(mut self, file: T) -> goblin::error::Result<()> {
         use goblin::elf::section_header::{SHN_LORESERVE, SHN_XINDEX};
         let mut file = BufWriter::new(file);
 
-        let needs_extended_symtab = self.nsections >= SHN_LORESERVE.into();
-        if needs_extended_symtab {
-            self.add_extended_symtab();
-        }
-
         /////////////////////////////////////
         // Compute Offsets
         /////////////////////////////////////
-        let sizeof_symtab = (self.symbols.len() + self.special_symbols.len() + self.sections.len())
-            * Symbol::size(self.ctx.container);
+        let symbol_count = self.symbols.len() + self.special_symbols.len() + self.sections.len();
+        let sizeof_symtab = symbol_count * Symbol::size(self.ctx.container);
```

Comment:
I'm really confused why these `X::size` calls changed; why use this here, and not below? Please change them back to their explicit versions.